### PR TITLE
update speculative decoding notebook

### PR DESCRIPTION
**Pull request by pavel-esir**
Update notebook in accordance with c++ sample https://github.com/openvinotoolkit/openvino.genai/pull/288

CVS-132865, https://github.com/openvinotoolkit/openvino.genai/issues/142

### CODE REVIEW

File: notebooks/266-speculative-sampling/266-speculative-sampling.ipynb [hunk 1]
```diff
@@ -23,8 +23,6 @@
    "id": "f97c435a",
```
Comment (eaidova) [inline notebooks/266-speculative-sampling/266-speculative-sampling.ipynb:23]: <p>are you sure that cleanup should be here?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1838/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (eaidova) [inline notebooks/266-speculative-sampling/266-speculative-sampling.ipynb:23]: <pre class="ql-syntax" spellcheck="false">Line #24.        main_ov_model = OVModelForCausalLM.from_pretrained(main_model_id, device=current_device, from_transformers=True)
</pre><p>from_transformers flag is deprecated, could you please replace with export=True? Also as your code below compile models without optimum, this model classes instances used only for models conversion, right?</p><p> </p><p>I think maybe better to use optimum-cli for export models if they are not exists and them load model from directory like in code below? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1838/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (pavel-esir) [reply notebooks/266-speculative-sampling/266-speculative-sampling.ipynb:23]: <p>replaced with optimum-cli. Now looks better. Also replaced with 7b, now runs faster and less ram is used.</p>

Comment (pavel-esir) [reply notebooks/266-speculative-sampling/266-speculative-sampling.ipynb:23]: <p>removed cleanup</p>

Comment (eaidova) [inline notebooks/266-speculative-sampling/266-speculative-sampling.ipynb:23]: <pre class="ql-syntax" spellcheck="false">Line #2.        !optimum-cli export openvino --model $main_model_id --weight-format fp16 $main_model_path
</pre><p>int8 or int4 will not be suitable here (required to install nncf only)?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1838/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (pavel-esir) [reply notebooks/266-speculative-sampling/266-speculative-sampling.ipynb:23]: i tried with all combinations and there are some problems. Surprisingly speculative decoding in int4/int8 cases takes more time:
image

I tried to take a look to outputs, and found out that there is a strange German word Unterscheidung

>Text = Alan Turing was a British mathematician, computer scientist, and logician who made significant contributions to the development of computer science, artificial intelligence, and cryptography. **Unterscheidung** between Turing machines and Turing degrees. Turing machines and Turing degrees are two.......

I suspect that this difference is causing problems, draft and main start to generate different outputs and therefore draft cannot assist the main network anymore. If i generate lees tokens before, then times are ok, and speculative is indeed faster. 
**Short sentence before German words appear**
image

Accuracy issues with int8/int4 are visible with other prompts as well, this strange German word still appears
**Different prompt int4**
image

Let's keep it fp16 and investigate why int4/int8 fails

[the remark this answers is not in the document]
Comment (pavel-esir) [reply notebooks/266-speculative-sampling/266-speculative-sampling.ipynb:23]: done

**Comment by eaidova** [comment]
@pavel-esir please retarget changes to master branch (not main)

**Comment by pavel-esir** [comment]
> @pavel-esir please retarget changes to master branch (not main)

done

**Review by eaidova — APPROVED** [review]

**Pull request merged**